tests/custom_lookups/tests.py

```diff
@@ -647,3 +671,61 @@ def test_subquery_usage(self):
                 id__in=Author.objects.filter(age__div3=2)
             )
             self.assertSequenceEqual(qs, [a2])
+
+
+class RegisterLookupTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.obj1 = Company.objects.create(name="apple", place="france")
+        cls.obj2 = Company.objects.create(name="atom", place="bangladesh")
+        cls.obj3 = Company.objects.create(name="cherry", place="finland")
+        cls.obj4 = Company.objects.create(name="all", place="fig")
+
+    def test_register_unregister_instance_lookup(self):
+        with register_lookup(Company.name.field, CustomStartsWith):
+            self.assertEqual(
+                Company.name.field.instance_lookups, {"sw": CustomStartsWith}
+            )
+        self.assertEqual(Company.name.field.instance_lookups, {})
+
+    def test_get_lookup_same_field(self):
+        with register_lookup(models.CharField, CustomStartsWith):
+            self.assertEqual(Company.name.field.get_lookup("sw"), CustomStartsWith)
+            self.assertEqual(Company.place.field.get_lookup("sw"), CustomStartsWith)
+
+    def test_get_lookup_specific_instance(self):
+        with register_lookup(Company.place.field, CustomStartsWith):
+            self.assertEqual(Company.name.field.get_lookup("sw"), None)
+            self.assertEqual(Company.place.field.get_lookup("sw"), CustomStartsWith)
+
+    def test_class_lookup_with_filter(self):
+        with register_lookup(models.CharField, CustomStartsWith):
+            self.assertSequenceEqual(
```

Comment:
Use `self.assertCountEqual()` when ordering is not specified and we have more than one expected result.

Reply:
> Use `self.assertCountEqual()` when ordering is not specified and we have more than one expected result.

Sure👍